Source/BugsnagConfiguration.m

```diff
@@ -199,6 +200,20 @@ - (void)setAppVersion:(NSString *)newVersion {
     }
 }
 
+@synthesize apiKey = _apiKey;
+
+- (NSString *)apiKey {
+    return _apiKey;
+}
+
+- (void)setApiKey:(NSString *)apiKey {
+    if (apiKey != nil) {
```

Comment:
`if (apiKey.length > 0)` would be more comprehensive here and catch both the nil and the empty string case.